### DISCUSSION TITLE
increment_build_number should not modify plist files

### DISCUSSION THREAD

**Discussion by ir-fuel** [fastlane/fastlane#17035] [Ideas] [answered]
I noticed that using `increment_build_number` modifies `info.plist` directly.

The thing is that normally that plist file contains: `$(CURRENT_PROJECT_VERSION)` as the build number, which is a value configured inside the project settings

![Screenshot 2020-08-11 at 12 33 32](https://user-images.githubusercontent.com/1573715/89887602-e29e6a00-dbce-11ea-9aa8-3e5663a3da41.jpg)

The downside of editing the plist directly using `increment_build_number` is that the value shown above is no longer in sync with the value that's actually used. Wouldn't it be better to just edit the value inside the `xcodeproj` file and not touch the plist? I understand it's currently working differently but maybe expose a parameter that would allow us to switch between both ways of modifying the build number could solve that?

**Comment by joshdholtz** [comment, marked as answer]
I completely agree with this! `$(CURERNT_PROJECT_VERSION)` is used/referenced in `get_version_number` but... `increment_build_number`, `get_build_number`, and `increment_version_number` all need to get updated 😇 

We should definitely get this changed 👍
